core/adapters/multiply.go

```diff
@@ -2,33 +2,34 @@ package adapters
 
 import (
 	"encoding/json"
-	"math/big"
 
-	"chainlink/core/utils"
+	"github.com/shopspring/decimal"
 )
 
 // Multiply holds the a number to multiply the given value by.
 type Multiply struct {
-	Times *big.Float `json:"-"`
+	Times decimal.Decimal `json:"-"`
 }
 
 type jsonMultiply struct {
-	Times *utils.BigFloat `json:"times,omitempty"`
+	Times decimal.Decimal `json:"times,omitempty"`
 }
 
 // MarshalJSON implements the json.Marshal interface.
 func (ma Multiply) MarshalJSON() ([]byte, error) {
-	jsonObj := jsonMultiply{Times: (*utils.BigFloat)(ma.Times)}
+	jsonObj := jsonMultiply{Times: ma.Times}
 	return json.Marshal(jsonObj)
 }
 
 // UnmarshalJSON implements the json.Unmarshal interface.
 func (ma *Multiply) UnmarshalJSON(buf []byte) error {
-	var jsonObj jsonMultiply
+	jsonObj := jsonMultiply{
+		decimal.NewFromInt(1), // default value
+	}
```

Comment:
What does setting a default value fix?  Unmarshalling JSON is prob not a good place to set these sorts of values

Reply:
Good question - it doesn't actually fix anything, but maintains the functionality of the previous adapter. Previously, if we wrote a job spec like this...

```
{
  "type": "multiply",
  "params": {}
},
```
Then we would assume the intended functionality was equivalent to...
```
{
  "type": "multiply",
  "params": {
    "times": 1
  }
},
```
(AKA do nothing...)

On the go side, this worked because we included this conditional that said something like
```go
if ma.Times != nil {	
  i.Mul(i, ma.Times)	
}
```

But since converting to `decimal.Decimal`, it no longer has the option to be a nil value and so if we don't include a "times" param in the job spec then the default value will be set to 0 :grimacing: .

Reply:
I think a better solution is probably just to require that we have a "times" param in the first place, instead of allowing incomplete multiply adapters... I I'm not sure if there is a use case for them.

Reply:
Its just setting default values like that in the json unmarshaler is unexpected.  Its outside of the JSON encoding domain.  Maybe a constructor or using a reference to decimal.Decimal.

But as suggested below, the utils.BigFloat was used for a reason

Reply:
Okay I removed the default, it definitely feels like an pattern now. The repercussions of omitting the "times" param will now be that the multiply adapter multiplies by 0. I think this will be a rare edge case and putting the responsibility on the job creator to do it right seems fair to me.